addons/controllers/clusterbootstrap_controller.go

```diff
@@ -213,6 +208,11 @@ func (r *ClusterBootstrapReconciler) reconcileNormal(cluster *clusterapiv1beta1.
 		return ctrl.Result{}, err
 	}
 
+	remoteClient, err := util.GetClusterClient(r.context, r.Client, r.Scheme, clusterapiutil.ObjectKey(cluster))
+	if err != nil {
+		return ctrl.Result{Requeue: true}, fmt.Errorf("failed to get remote cluster client: %w", err)
```

Comment:
Add a delay with `RequeueAfter: constants.RequeueAfterDuration`. At this point the cluster is "ready" based on cluster api but the api-server may need a while to accept requests.